### ISSUE TITLE
 ROSBagException should call the base constructor

### ISSUE THREAD

**Issue by lukehansen** [ros/ros_comm#1209]
In python 2.7 there is a subtle bug with unpickling custom exceptions; this was fixed in 3.3, see https://bugs.python.org/issue1692335.  As a workaround for 2.7 we can re-define these exceptions as below.

```java
class ROSBagException(Exception):
    """
    Base class for exceptions in rosbag.
    """
    pass
 
class ROSBagFormatException(ROSBagException):
    """
    Exceptions for errors relating to the bag file format.
    """
    pass

class ROSBagUnindexedException(ROSBagException):
    """
    Exception for unindexed bags.
    """
    def __init__(self, *args):
        ROSBagException.__init__(self, 'Unindexed bag')
```

For context, I came across this issue when a `ROSBagUnindexedException` was raised in a `multiprocessing.Pool.map` thread.  It throws the `__init__` exception and hangs, as described here: https://lists.gt.net/python/bugs/1025933

**Comment by dirk-thomas** [comment]
Can you please provide a pull request with the proposed patch against the current default branch (`lunar-devel`). Thanks.

**Comment by lukehansen** [comment]
Sure, though it's not really a ros_comm bug and has been patched in python 3.3.  PR is here: https://github.com/ros/ros_comm/pull/1210
